### PR TITLE
WIP: Add a advanced compiler command 'ctags'

### PR DESCRIPTION
**Pull request by data-man**
This command generates a tags file for a source file.

## Task list
- [x] Generate a tags file
- [ ] Config file?

### CODE REVIEW

File: doc/advopt.txt [hunk 1]
```diff
@@ -7,7 +7,8 @@ Advanced commands:
   //rst2html                convert a reStructuredText file to HTML
   //rst2tex                 convert a reStructuredText file to TeX
   //jsondoc                 extract the documentation to a json file
-  //jsondoc2                extract documentation to a json file (uses doc2)
+  //jsondoc2                extract the documentation to a json file (uses doc2)
+  //tags                    extract a tags to a tags file
```
Comment (Araq) [inline doc/advopt.txt]: What?

File: compiler/docgen.nim [hunk 2]
```diff
@@ -608,6 +608,53 @@ proc generateJson*(d: PDoc, n: PNode) =
       generateJson(d, lastSon(n.sons[0]))
   else: discard
 
+proc genTagsItem(d: PDoc, n, nameNode: PNode, k: TSymKind): string =
+  var
+    name = getName(d, nameNode)
+    comm = $genRecComment(d, n)
```
Comment (Araq) [inline compiler/docgen.nim]: 'comm' is not used.

File: doc/advopt.txt (hunk 1 above)
Comment (data-man) [reply doc/advopt.txt]: But how?
"Create a tags file" ?

**Comment by dom96** [comment]
A "tags file"? What is it?

**Comment by data-man** [comment]
@dom96 
https://en.wikipedia.org/wiki/Ctags
https://en.wikipedia.org/wiki/Ctags#Tags_file_formats

**Comment by dom96** [comment]
So call it ``ctags``?

**Comment by data-man** [comment]
No, ctags is app and .tags is a format.

**Comment by dom96** [comment]
From wikipedia:

> The tags file is normally named "tags"

So it should actually be called "tags", not ".tags".

But I wasn't referring to the filename, I was referring to the command. You should run ``nim ctags`` to get it, not ``nim tags``.

**Comment by data-man** [comment]
> So it should actually be called "tags", not ".tags".

It is file extension.
The current implementation creates a file named sourceFileName.tags
This can be changed, of course.
And file-tags are created with a symbol names only.
So I want to add the nimtags.cfg file for configuring various tags generating parameters (comments, # line, # column, etc.).

> You should run nim ctags to get it, not nim tags.

Ok.

**Pull request merged**